internal/awsservice/imds.go

```diff
@@ -11,8 +11,20 @@ import (
 	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
 )
 
-// TODO: Refactor Structure and Interface for more easier follow that shares the same session
 func GetInstanceId() string {
+	return GetImdsMetadata().InstanceID
+}
+
+func GetImageId() string {
+	return GetImdsMetadata().ImageID
+}
+
+func GetInstanceType() string {
+	return GetImdsMetadata().InstanceType
+}
+
+// TODO: Refactor Structure and Interface for more easier follow that shares the same session
+func GetImdsMetadata() *imds.GetInstanceIdentityDocumentOutput {
```

Comment:
We should cache the results of the metadata call so we don't ping IMDS a lot.

Reply:
Added a variable to cached the results. Thanks

Reply:
Do we plan on calling this multiple times within the same test execution, that is needs to be cached?